modules/codegen-plugin/src/smithy4s/codegen/Smithy4sCodegenPlugin.scala

```diff
@@ -83,25 +83,26 @@ object Smithy4sCodegenPlugin extends AutoPlugin {
 
   override def projectConfigurations: Seq[Configuration] = Seq(Smithy4s)
 
-  override lazy val projectSettings =
-    Seq(
-      Compile / smithy4sInputDir := (Compile / sourceDirectory).value / "smithy",
-      Compile / smithy4sOutputDir := (Compile / sourceManaged).value,
-      Compile / smithy4sOpenapiDir := (Compile / resourceManaged).value,
-      Compile / smithy4sCodegen := cachedSmithyCodegen(Compile).value,
-      Compile / smithy4sCodegenDependencies := List.empty: @annotation.nowarn,
-      Compile / sourceGenerators += (Compile / smithy4sCodegen).map(
-        _.filter(_.ext == "scala")
-      ),
-      Compile / resourceGenerators += (Compile / smithy4sCodegen).map(
-        _.filter(_.ext != "scala")
-      ),
-      cleanFiles += (Compile / smithy4sOutputDir).value,
-      Compile / smithy4sModelTransformers := List.empty
-    )
+  // Use this with any configuration to enable the codegen in it.
+  def defaultSettings(config: Configuration) = Seq(
+    config / smithy4sInputDir := (config / sourceDirectory).value / "smithy",
+    config / smithy4sOutputDir := (config / sourceManaged).value,
+    config / smithy4sOpenapiDir := (config / resourceManaged).value,
+    config / smithy4sCodegen := cachedSmithyCodegen(config).value,
+    config / smithy4sCodegenDependencies := List.empty: @annotation.nowarn,
+    config / sourceGenerators += (config / smithy4sCodegen).map(
+      _.filter(_.ext == "scala")
+    ),
+    config / resourceGenerators += (config / smithy4sCodegen).map(
+      _.filter(_.ext != "scala")
+    ),
+    cleanFiles += (config / smithy4sOutputDir).value,
+    config / smithy4sModelTransformers := List.empty
+  )
 
-  private def untupled[A, B, C](f: ((A, B)) => C): (A, B) => C = (a, b) =>
-    f((a, b))
+  override lazy val projectSettings =
+    defaultSettings(Compile) ++
+      defaultSettings(Test)
```

Comment:
> not sure we want this for `Test` too by default. It means that the compilation of tests will try to generate stuff from smithy, meaning it'll load a model, which takes computations.
> 
> I'd accept it if it was accompanied by a change of logic that'd verify that `CodegenArgs` contains only default values (in which case it would circumvent all smithy loading logic.

I don't understand the issue here. I guess the use case is that you _want_ code generation for this smithy definition only in `Test`. Personally, I have never needed to do this, but I can understand why one would need it.

Reply:
I don't like it before having it set by default means more computations before you're able to run tests, because even if you don't have specs in the test scope, smithy4s will run codegen when compiling the test scope.

Reply:
like, my usecase is this: I'm authoring a library/tool that has some SchemaVisitors, and I want to test it with some real smithy4s-generated models. I don't want these in the jars I publish, so I put that in my test sources.